File: docs/data-structures/linked-lists/code/cpp/linked_list_examples.cpp
#include <iostream>

struct Node {
    int value;
    Node* next;
};

void insert_front(Node*& head, int value) {
    Node* node = new Node{value, head};
    head = node;
}

void traverse(const Node* head) {
    const Node* current = head;
    while (current != nullptr) {
        std::cout << current->value << ' ';
        current = current->next;
    }
    std::cout << '\n';
}

int main() {
    Node* head = nullptr;
    insert_front(head, 30);
    insert_front(head, 20);
    insert_front(head, 10);

    traverse(head);

    while (head != nullptr) {
        Node* next = head->next;
        delete head;
        head = next;
    }

    return 0;
}

File: docs/data-structures/linked-lists/code/python/linked_list_examples.py
class Node:
    def __init__(self, value: int, next_node: "Node | None" = None) -> None:
        self.value = value
        self.next = next_node


def insert_front(head: Node | None, value: int) -> Node:
    return Node(value, head)


def traverse(head: Node | None) -> None:
    current = head
    while current is not None:
        print(current.value, end=" ")
        current = current.next
    print()


def main() -> None:
    head: Node | None = None
    head = insert_front(head, 30)
    head = insert_front(head, 20)
    head = insert_front(head, 10)

    traverse(head)


if __name__ == "__main__":
    main()